### DISCUSSION TITLE
How to debug/diagnose endless loop/cycle in M2E Builder

### DISCUSSION THREAD

**Discussion by guw** [eclipse-m2e/m2e-core#619] [Q&A]
Is there some log output I can turn on? I have a workspace where M2E builder is constantly looping over a set of projects. It's not stopping.

**Comment by laeubi** [comment]
You might get some ideas here:
- https://github.com/eclipse-m2e/m2e-core/issues/123

**Comment by davidmichaelkarr** [comment]
I've run into this when I had multiple executions of a plugin that generates files, and each execution writes data into directories that are referenced by other executions.  For instance, our enterprise has a lot of services that use the jaxb2 plugin, but most developers don't realize that each of those executions will write to the same base directory, even if the resulting "leaf" directories are different. To avoid this, for each one I specify a "generateDirectory" value that adds one directory to the value, using a name specific to that execution. You also have to ensure that "forceRegenerate" is false.
